### PR TITLE
Web Workers

### PR DESCRIPTION
**Pull request by loretoparisi**
I have added Web Workers to read files. This will prevent the browser window to be stuck when loading huge dataset files, while parsing the file:

(index.html)
```javascript
var fileworker = new Worker('worker.js');
    fileworker.onmessage = function(e) {
        var datasetsList=e.data;
        whendone(datasetsList);
    };
    fileworker.onerror = function(e) {
      console.error('ERROR: Line ', e.lineno, ' in ', e.filename, ': ', e.message);
    };
    var readFileAsync = function() {
      fileworker.postMessage(fileupload.files);
    }
    fileupload.addEventListener('change', readFileAsync);
```

where the `whendone` callback still handles the visualization in the page, while the csv file parsing via the `d3.csvParse` api is done by the web worker in `worker.js` as well as the file contents reading:

(worker.js)
```javascript
self.addEventListener('message', function (e) {
    var files = e.data;
    readFiles(files, function (datasetsList) { // done
        postMessage(datasetsList);
    });
}, false);
```

where the `readFiles` has been modified in the synchronous way since we are in the asynchronous web worker processing:

```javascript
var readFiles = function (files,whendone) {
    var count = files.length;
    var datasetsList = [];
    var readFile = function (file) {
        var reader = new FileReaderSync();
        var result=reader.readAsBinaryString(file);
        var parsedData = d3.csvParse(result);
            parsedData.forEach(function (row) {
                parsedData.columns.forEach(function (column) {
                    if (!row[column]) {
                        delete row[column];
                    }
                    var coercedNum = +row[column];
                    if (!isNaN(coercedNum)) {
                        row[column] = coercedNum;
                    }
                });
            });
            datasetsList.push({ data: parsedData, name: file.name });
            if (!--count) {
                whendone(datasetsList);
            }
        
    };

    for (var i = 0; i < count; i++) {
        readFile(files[i]);
    }
}
```

### CODE REVIEW

File: worker.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/**
+ * @author Loreto Parisi (loretoparisi at gmail dot com)
```
Comment (jameswex) [inline worker.js]: we avoid author tags in the source in our repo. github's history includes more detailed authorship information already. would you be ok removing this?

**Comment by loretoparisi** [comment]
@jameswex you are welcome, it makes sense, it's ok!

**Comment by jameswex** [comment]
feel free to push another version that removes the author line and i can submit it.

**Comment by loretoparisi** [comment]
@jameswex done!

**Pull request merged**